src/pe/optional_header.rs

```diff
@@ -6,52 +8,125 @@ use crate::pe::data_directories;
     pub base_of_data: u32,
 }
 
+/// Convenience constant for `core::mem::size_of::<StandardFields32>()`.
 pub const SIZEOF_STANDARD_FIELDS_32: usize = 28;
 
-/// standard 64-bit COFF fields
+/// Standard 64-bit COFF fields (for `PE32+`).
+///
+/// In `winnt.h`, this is a subset of [`IMAGE_OPTIONAL_HEADER64`](https://learn.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-image_optional_header64).
+///
+/// * For 32-bit version, see [`StandardFields32`].
+/// * For unified version, see [`StandardFields`].
 #[repr(C)]
 #[derive(Debug, PartialEq, Copy, Clone, Default, Pread, Pwrite, SizeWith)]
 pub struct StandardFields64 {
+    /// See docs for [`StandardFields::magic`](crate::pe::optional_header::StandardFields::magic).
     pub magic: u16,
+    /// See docs for [`StandardFields::major_linker_version`].
     pub major_linker_version: u8,
+    /// See docs for [`StandardFields::minor_linker_version`].
     pub minor_linker_version: u8,
+    /// See docs for [`StandardFields::size_of_code`].
     pub size_of_code: u32,
+    /// See docs for [`StandardFields::size_of_initialized_data`].
     pub size_of_initialized_data: u32,
+    /// See docs for [`StandardFields::size_of_uninitialized_data`].
     pub size_of_uninitialized_data: u32,
+    /// See docs for [`StandardFields::address_of_entry_point`].
     pub address_of_entry_point: u32,
+    /// See docs for [`StandardFields::base_of_code`].
     pub base_of_code: u32,
 }
 
+/// Convenience constant for `core::mem::size_of::<StandardFields64>()`.
 pub const SIZEOF_STANDARD_FIELDS_64: usize = 24;
 
-/// Unified 32/64-bit COFF fields
+/// Unified 32/64-bit standard COFF fields (for `PE32` and `PE32+`).
+///
+/// Notably, a value of this type is a member of
+/// [`goblin::pe::optional_header::OptionalHeader`](crate::pe::optional_header::OptionalHeader),
+/// which in turn represents either
+/// * [`IMAGE_OPTIONAL_HEADER32`](https://learn.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-image_optional_header32); or
+/// * [`IMAGE_OPTIONAL_HEADER64`](https://learn.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-image_optional_header64)
+///
+/// from `winnt.h`, depending on the value of [`StandardFields::magic`].
+///
+/// ## Position in PE binary
+///
+/// Standard COFF fields are located at the beginning of the [`OptionalHeader`] and before the
+/// [`WindowsFields`].
+///
+/// ## Related structures
+///
+/// * For 32-bit version, see [`StandardFields32`].
+/// * For 64-bit version, see [`StandardFields64`].
 #[derive(Debug, PartialEq, Copy, Clone, Default)]
 pub struct StandardFields {
+    /// The state of the image file. This member can be one of the following values:
+    ///
+    /// * [`IMAGE_NT_OPTIONAL_HDR32_MAGIC`].
+    /// * [`IMAGE_NT_OPTIONAL_HDR64_MAGIC`].
+    /// * [`IMAGE_ROM_OPTIONAL_HDR_MAGIC`].
+    #[doc(alias = "Magic")]
     pub magic: u16,
+    /// The major version number of the linker.
+    #[doc(alias = "MajorLinkerVersion")]
```

Comment:
i don't think this aliases anything unless i'm misunderstanding what `doc(alias` does

Reply:
no i was wrong :)

Reply:
Most likely, you were misunderstanding. The link to the explanation provided above.